### PR TITLE
SONAR-6822 langName is not returned

### PR DESCRIPTION
**Pull request by julienlancelot**
/api/rules/search?f=langName is now returning langName
/api/rules/search?f=lang doest not return anymore langName

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java [hunk 1]
```diff
@@ -317,6 +318,11 @@ private static void setInternalKey(Rules.Rule.Builder ruleResponse, Rule ruleDoc
   private void setLanguage(Rules.Rule.Builder ruleResponse, Rule ruleDoc, Set<String> fieldsToReturn) {
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java]: Can be static.

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java [hunk 2]
```diff
@@ -317,6 +318,11 @@ private static void setInternalKey(Rules.Rule.Builder ruleResponse, Rule ruleDoc
...
     if (shouldReturnField(fieldsToReturn, RuleNormalizer.RuleField.LANGUAGE) && ruleDoc.language() != null) {
       ruleResponse.setLang(ruleDoc.language());
+    }
+  }
+
+  private void setLanguageName(Rules.Rule.Builder ruleResponse, Rule ruleDoc, Set<String> fieldsToReturn) {
+    if (shouldReturnField(fieldsToReturn, "langName") && ruleDoc.language() != null) {
       Language language = languages.get(ruleDoc.language());
       if (language != null) {
         ruleResponse.setLangName(language.getName());
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java]: I suggest to return the lang key when the name is not available.

**Comment by simonbrandhof** [comment]
Two feedbacks then LGTM

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java]: Right !

File: server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java]: It will change existing behaviour but I think it's a good idea.

Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapping.java]: of course it's a good idea :-)

**Pull request merged**